core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -71,22 +71,32 @@ import { SubtitledHtml } from 'domain/exploration/subtitled-html.model';
 
 
 export interface InteractionBackendDict {
+  // A null 'defaultOutcome' indicated a that this interaction is
+  // an EndExploration interaction.
   'default_outcome': OutcomeBackendDict | null;
   'answer_groups': readonly AnswerGroupBackendDict[];
   'confirmed_unclassified_answers': readonly InteractionAnswer[];
   'customization_args': InteractionCustomizationArgsBackendDict;
   'hints': readonly HintBackendDict[];
   'id': string;
+  // A null 'solution' indicates that this Interaction does not have a hint
+  // or there is a hint, but no solution. A new interaction is initialised with
+  // null 'solution' and stays null until the first hint with solution is added.
   'solution': SolutionBackendDict | null;
 }
 
 export class Interaction {
   answerGroups: AnswerGroup[];
   confirmedUnclassifiedAnswers: readonly InteractionAnswer[];
   customizationArgs: InteractionCustomizationArgs;
+  // A null 'defaultOutcome' indicated a that this interaction is
+  // an EndExploration interaction.
   defaultOutcome: Outcome | null;
   hints: Hint[];
   id: string;
+  // A null 'solution' indicates that this Interaction does not have a hint
+  // or there is a hint, but no solution. A new interaction is initialised with
```

Comment:
Comments are fine at the top once, we can remove these

Reply:
done